src/routes/+page.svelte

```diff
@@ -1,21 +1,20 @@
-<script context="module" lang="ts">
-  import type * as prismicT from "@prismicio/types"
+<script lang="ts">
   import { SliceZone } from "$lib";
-  import {RichText, CodeSnippet} from "./slices"
+  import { RichText, CodeSnippet } from "$lib/slices"
+  // import type { PageData } from './$types'
 
   const components = {
     rich_text: RichText,
     code_snippet: CodeSnippet
   }
-</script>
 
-<script lang="ts">
-  export let document:prismicT.PrismicDocument
+  export let data: any
```

Comment:
I recommend we type `data` using `@prismicio/types` even if this is just a development tool. Typing the data will make it easier to play with and verify the data is compatible with components it interacts with.

Reply:
@angeloashmore If I put the Slice and Custom Type types in a separate file, what's the convention for naming and locating those files?

Reply:
I don't know if there's a SvelteKit-specific convention, but in general TypeScript projects, a `types.ts` file usually works well.

For example, you could colocate a `PageDocument` type like this:

```ts
// src/routes/types.ts

import * as prismicT from "@prismicio/types";

export type PageDocument = prismicT.PrismicDocument<{
  title: prismicT.RichTextFIeld
}, "page">
```

For simple types, you could also drop the type directly in the file where it is used. I would personally go with that option since the demo app isn't a project that needs to scale to a large number of files and developers. Totally up to you though! Do whatever you think works best.

Alternatively, you could use `prismic-ts-codegen` to generate types from your example repository. This is probably overkill.

Reply:
I saw you created a `src/types` directory. That works well, too.

I recommend changing the types from `interfaces` to `types` and use `@prismicio/types`'s type parameters when possible. This lets us change how e.g. `SliceZone` works while retaining access to your Slice types.

```diff
  import type * as prismicT from "@prismicio/types";
  import type { CodeSnippetSlice } from "./CodeSnippetSlice.type";
  import type { RichTextSlice } from "./RichTextSlice.type";

+ export type PageData = prismicT.PrismicDocument<{
+ 	body: prismicT.SliceZone<CodeSnippetSlice | RichTextSlice;
+ }>

- export interface PageData extends prismicT.PrismicDocument {
- 	document: {
- 		data: {
- 			body: [CodeSnippetSlice | RichTextSlice];
- 		};
- 	};
- }
```

One last nit: "Data" is a generic term that doesn't provide much meaning. I recommend naming it something that more closely represents what it is: `PageDocument`.

Reply:
I have updated the Slice types and the Page type, and I think I'm making better use of `@prismicio/types` 🤞